src/pebble.py

```diff
@@ -259,6 +262,11 @@ def change_config(charm_state: CharmState, container: ops.model.Container) -> No
         if charm_state.datasource:
             logger.info("Synapse Stats Exporter enabled.")
             replan_stats_exporter(container=container, charm_state=charm_state)
+        if charm_state.synapse_config.enable_irc_bridge:
```

Comment:
Regarding that refactor that should be done in Synapse to prevent restarts...Putting this here wouldn't make IRC Bridge restart for unrelated Synapse configuration changes?

Reply:
I feel that it shouldn't (per [replan doc](https://ops.readthedocs.io/en/latest/index.html#ops.Container.replan)):

```text
Replan all services: restart changed services and start startup-enabled services.
```

It should not restart a service that has not changed. The whole flow can be improved, though now it shouldn't have the effect of restarting the service or regenerating a file if it has been already generated. The steps are:

* is irc bridge enabled in config
* if so, check whether we have a db relation
* if yes, generate a config file for irc if one does not already exist
* generate an app registration file if one does not already exist
* restart the service if the service has changed